crates/emitter/src/scope_notes.rs

```diff
@@ -62,9 +62,36 @@ impl ScopeNoteList {
         ScopeNoteIndex::new(note_index)
     }
 
+    fn get_scope_hole_gcthing_index(&self, reference_scope_index: ScopeNoteIndex) -> GCThingIndex {
+        if reference_scope_index.index >= 1 {
+            return self
+                .notes
+                .get(reference_scope_index.index - 1)
+                .unwrap()
+                .index;
+        }
+        GCThingIndex::new(0)
```

Comment:
this isn't guaranteed to point the global scope.
please return `InstructionWriter::body_scope_index` somehow,

or (to leave this to other issue) add an assertion with explanation to `InstructionWriter::enter_global_scope`,
so that once another GC thing gets 0-th index, it fails and we can revisit here.